xarray/core/merge.py

```diff
@@ -365,6 +365,24 @@ def merge_data_and_coords(data, coords, compat='broadcast_equals',
     return merge_core(objs, compat, join, explicit_coords=explicit_coords)
 
 
+def assert_valid_explicit_coords(variables, explicit_coords):
+    """Validate explicit coordinate names/dims.
+
+    Raise a MergeError if an explicit coord shares a name with a dimension
+    but is comprised of arbitrary dimensions.
+    """
+    var_dims = []
+    for name, var in variables.items():
+        if name not in explicit_coords:
+            var_dims.extend(var.dims)
+    for coord_name in explicit_coords:
```

Comment:
I think you're missing this as a function argument

Reply:
`explicit_coords` is a function argument.

Reply:
I was misreading something else, nevermind